hexa/workspaces/schema/mutations.py

```diff
@@ -122,57 +122,61 @@ def resolve_invite_workspace_member(_, info, **kwargs):
             slug=input["workspaceSlug"]
         )
         try:
-            # If the user already exists, we add it to the workspace
             user = User.objects.get(email=input["userEmail"])
-            workspace_membership = WorkspaceMembership.objects.create_if_has_perm(
-                principal=request.user,
-                workspace=workspace,
-                user=user,
-                role=input["role"],
-            )
+            # If the user already exists, check if his not already a member of the workspace
+            if WorkspaceMembership.objects.filter(
+                user__email=input["userEmail"], workspace=workspace
```

Comment:
Since we found the user, let's use it to look for the membership
```suggestion
                user=user, workspace=workspace
```